docs/src/pages/components/data-grid/overview/overview.md

```diff
@@ -66,7 +66,7 @@ We provide three options:
 
 - Built with and exclusively for React ⚛️
 - High performance 🚀
-- Lightweight; less than [30 kB](https://bundlephobia.com/result?p=@material-ui/data-grid) gzipped with as few dependencies as possible.
+- Lightweight; less than [55 kB](https://bundlephobia.com/result?p=@material-ui/data-grid) gzipped with as few dependencies as possible.
```

Comment:
This is not lightweight, this is bloated. I couldn't spot any major bump. It seems to be simply related to the features. Maybe we write them in a way that is particularly bundled size intensive. No idea

```suggestion
- Weight [55 kB](https://bundlephobia.com/result?p=@material-ui/data-grid) gzipped with as few dependencies as possible.
```

Reply:
Maybe we altogether remove this mentioning of 55kb bundle size from `Features` if it's risky to mention which may lead to less/no users. Better than mentioning false size (user doesn't know we forgot to update doc). When we act on #924 we can specify it as a `good` feature of isolation.  
Also is data-grid tree-shakeable?

Reply:
> Also is data-grid tree-shakeable

Yes, to some extent. For instance, because we didn't bundle the Toolbar, it's. I think that ideally with #924, we would create a public hook API developers can use, with a plugin system.

Why not remove the weight 👍 

```suggestion
```

Reply:
Removed.

Reply:
https://github.com/mui-org/material-ui-x/issues/924 is part of the beta so we need to start looking at it. I'm interested in picking it up.

Reply:
Actually no, I was wrong. It looks like tree-shaking doesn't work with our packages: 

<img width="629" alt="Screenshot 2021-04-22 at 20 00 11" src="https://user-images.githubusercontent.com/3165635/115764016-5bcc9e80-a3a5-11eb-8227-b9b190f5c628.png">

This looks like a quick win. We have been careless about this aspect. https://github.com/mui-org/material-ui/blob/4d7fb7c197f6bc345e9705d45dfb5e03bef70075/packages/material-ui/package.json#L78

The previous investigation regarding bundle size opportunities: https://github.com/mui-org/material-ui-x/pull/954.

@DanailH It depends on what we want to optimize for. If bundle size is our biggest interest, we should start with bundle size monitoring at each PR. Being able to continuously measure changes is paramount. We can only improve what you can measure. Same reason why writing tests is critical when solving a complex problems.

Reply:
I spend half an hour diving into why this happens. There are more issues than I thought. I have updated https://github.com/mui-org/material-ui-x/issues/18 with the problems and solutions. For instance, all the developers bundle all the locales 😱.